src/feasibility_checker.jl

```diff
@@ -0,0 +1,141 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+using LinearAlgebra
+
+"""
+    primal_feasibility_report(
+        model::Model,
+        [point::AbstractDict{VariableRef,Float64}];
+        atol::Float64 = 1e-8,
+    )::Union{Nothing,Dict{Any,Float64}}
+
+Check the primal feasibility of `model` at the point given by the dictionary
+`point` mapping variables to primal values. If a variable is not given in
+`point`, the value is assumed to be `0.0`.
+
+If the point is feasible, this function returns `nothing`. If infeasible, this
+function returns a dictionary mapping the constriant reference of each violated
+constraint to the distance it is from being feasible.
+
+To obtain `point` from a solution of a solved model, use:
+```julia
+point = Dict(v => value(v) for v in all_variables(model))
+```
+"""
+function primal_feasibility_report(
+    model::Model,
+    point::AbstractDict{VariableRef,Float64};
+    atol::Float64 = 1e-8,
```

Comment:
Does it make sense to have a default tolerance here? Any value we put here means that in some cases, a solver will return a solution that it thinks is feasible, and `primal_feasibility_report` will declare it infeasible. I would either force the user to provide a tolerance, or just return all infeasibilities greater than zero and let the caller decide how to filter them.